### DISCUSSION TITLE
How to correctly use (*pgtype.Map)SQLScanner?

### DISCUSSION THREAD

**Discussion by vladsavostin** [jackc/pgx#1936] [Q&A]
pgx lib has a nice examples with scaning into go slices, but them use `(*pgtype.Map)SQLScanner`. Standard libs like as database/sql and jmoiron/sqlx don't set this mapper before scanning, because this map and scanner are driver-specific.

**Comment by jackc** [comment]
Unfortunately, there is no simple way to bridge pgx's type system to database/sql without SQLScanner.

See https://github.com/jackc/pgx/issues/1779, https://github.com/jackc/pgx/issues/1781, https://github.com/jackc/pgx/issues/1662, and https://github.com/jackc/pgx/issues/1458 for some previous discussions and partial solutions / workarounds.
